Make fingerprint CLI consistent with predict CLI

hwpang opened the pull request:
## Description
Recently we have merged in https://github.com/chemprop/chemprop/pull/731, which changes the behavior of `--model-path` in predict.py to be the following

- It can take multiple locations of model (.pt) files and (.ckpt) checkpoints
- If a directory is given, search for all (.pt) models using the `find_models` helper function

The fingerprint.py has a same argument `--model-path` and a duplicate `find_models` helper function.

Instead of maintaining two functions doing the same thing, I move the `find_models` helper function into common.py. Both predict and fingerprint now use the same helper function code.

I change the `--model-path` in fingerprint.py to behave the same as predict.py, and change the help string to reflect the new behavior.

I also added the index corresponding to the model path to make bookkeeping easier for the user.

## Checklist
- [x] linted with flake8?
- [ ] (if appropriate) unit tests added?

shihchengli commented on chemprop/cli/fingerprint.py:
```diff
@@ -40,10 +45,12 @@ def add_args(cls, parser: ArgumentParser) -> ArgumentParser:
             help="Path to which predictions will be saved. If the file extension is .npz, they will be saved as a npz file, respectively. Otherwise, will save predictions as a CSV. The index of the model will be appended to the filename's stem. By default, predictions will be saved to the same location as '--test-path' with '_fps' appended, i.e., 'PATH/TO/TEST_PATH_fps_0.csv'.",
         )
         parser.add_argument(
+            "--model-paths",
```
We don't have this in predict.py yet. I would prefer to add it, but I'm not sure whether we should make `--model-paths` or `--model-path` the priority.

hwpang replied:
True, this PR should probably wait after your PR https://github.com/chemprop/chemprop/pull/919 is merged. What ever is decided in your PR, I will make the same changes accordingly for consistency.

shihchengli replied:
I will make some changes related to this flag in my PR.

shihchengli reviewed (commented):
Thanks for making this PR. I was wondering when we should put the arguments in common.py? I can see that `--model-path` would be used in both predict.py and fingerprint.py; should we move this into common.py? Similarly, this applies to `--target-columns`, but this flag in predict.py will be removed, as we will save the target names in the model file in the future. Do we want to move the arguments being used twice in train/pridict/hpopt/fingerprint into common.py?

hwpang commented:
> If we added it to common, then `--model-paths` would also appear when you do `chemprop train -h` which is probably confusing for users. It also forces the help message to be the same for `--model-paths` in both predict and fingerprint, which may be desirable, but currently the help message in predict specifically says it is the location of model files to use for prediction.
> 
> This does raise the question of what is the correct way to build an maintain a CLI parser. My thought is that eventually we will probably stop updating the CLI an only have new features available in modules. I think this because our development team is likely to shrink. If we do intend to phase out development of the CLI, then it makes sense to do what is the least work in the short term for the CLI an not worry about the long term. To me the least work is to duplicate code when it is not simple to import from other parts of the CLI. I think I would have near duplicate `--model-paths` code in predict and fingerprint, but have fingerprint import `find_models()` from `chemprop.cli.predict`.

@KnathanM Thanks, this makse sense to me. I've moved find_models back to chemprop.cli.predict and have fingerprint import from it.
